src/ui/chart-list.jsx

```diff
@@ -0,0 +1,241 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import CHARTS from '../data/charts.json';
+import LogicCalculation from '../services/logic-calculation';
+import LogicHelper from '../services/logic-helper';
+import Spheres from '../services/spheres';
+import TrackerState from '../services/tracker-state';
+
+import FoundAtTooltip from './found-at-tooltip';
+import Images from './images';
+import KeyDownWrapper from './key-down-wrapper';
+import Tooltip from './tooltip';
+
+class ChartList extends React.PureComponent {
+  static NUM_ROWS = 20;
+
+  mapChart(chart) {
+    const {
+      openedChartForIsland,
+      trackerState,
+      updateChartMapping,
+    } = this.props;
+
+    if (_.isNil(chart)) {
+      return null;
+    }
+
+    const itemCount = trackerState.getItemValue(chart);
+
+    const mappedIslandForChart = trackerState.getIslandForChart(chart);
+    const isChartMapped = !_.isNil(mappedIslandForChart);
+
+    const notInteractiveClassName = isChartMapped ? 'detail-not-interactive' : '';
+
+    let color;
+    if (isChartMapped) {
+      color = LogicCalculation.LOCATION_COLORS.CHECKED_LOCATION;
+    } else if (itemCount === 1) {
+      color = LogicCalculation.LOCATION_COLORS.AVAILABLE_LOCATION;
+    } else {
+      color = LogicCalculation.LOCATION_COLORS.UNAVAILABLE_LOCATION;
+    }
+
+    const updateChartMappingFunc = (event) => {
+      event.stopPropagation();
+      if (!isChartMapped) {
+        updateChartMapping(chart, openedChartForIsland);
```

Comment:
If we're mapping a chart and it hasn't been checked off yet in the chart list, this function should mark the chart as checked (by calling `incrementItem` on the chart item). Otherwise, the chart list becomes desynced from the mapping.